Language/Haskell/GhcMod/ImportedFrom.hs

```diff
@@ -552,7 +552,10 @@ refineVisibleExports getHaddockInterfaces exports = mapM f exports
             qexports       = qualifiedExports mexports -- e.g. ["base-4.8.2.0:GHC.Base.Just", ...]
         mVisibleExportsMap <- getVisibleExports getHaddockInterfaces pname
 
-        let visibleExportsMap = fromMaybe (throw $ GMEString $ "ImportedFrom: visible exports map is Nothing") mVisibleExportsMap
+        visibleExportsMap <- case mVisibleExportsMap of
```

Comment:
I think you misunderstood, there are three kinds of failures:

- A pure exception

```
do let foo = throw "whatever"
     return foo
```

- An excpetion where a monadic value should be

```
do throw "whatever"
```

- An ExceptT/MonadError failure

```
do throwError $ GMEString $ "whatever"
```


The differences are subtle but in general the first two kinds are annoying since you need to explicitly catch the exceptions whereas the last one just always gets returned as a `Left` by `runGhcModT`.

The difference between 1) and 2) is important though: `throw "whatever" >> return "foo"` will always crash when run, `let foo = throw "whatever" in return (foo, "bar")` will only crash when you touch the fst of that tuple! (lazyness and such)

The GhcModError stuff should always be returned via `throwError` where possible. All you really had to do was to replace `fromMaybe` with `liftMaybe` :)